src/main/kotlin/bsu/cc/schedule/Models.kt

```diff
@@ -0,0 +1,40 @@
+package bsu.cc.schedule
+
+import com.brein.time.timeintervals.intervals.IInterval
+import java.time.DayOfWeek
+import java.time.LocalDate
+import java.time.LocalTime
+
+
+data class Instructor(val firstName: String, val lastName: String)
+
+data class DateInterval(val startDate: LocalDate, val endDate: LocalDate)
+
+class ClassSchedule(val startTime: LocalTime,
+                    val endTime: LocalTime,
+                    val meetingDays: Set<DayOfWeek>,
+                    val meetingDates: DateInterval,
+                    val subject: String,
+                    val catalogNumber: String,
+                    val section: String,
+                    val room: String,
+                    val instructors: Set<Instructor>,
+                    val description: String
+                    ) : IInterval<Int> {
+
+    override fun compareTo(other: IInterval<*>?): Int {
+        val cmpStart = startTime.toSecondOfDay() - other?.getNormStart() as Int
+        if (cmpStart == 0) {
+            return endTime.toSecondOfDay() - other.getNormEnd() as Int
+        }
+        return cmpStart
+    }
+
+    override fun getUniqueIdentifier(): String = "[$startTime, $endTime]"
```

Comment:
Does this actually have to be completely unique in the tree or just unique to how it will be treated by the tree (IE: classes with the same start and end time are treated as the same thing)? If it does have to be unique to the tree, could we run into issues with classes that have the same start and end time? I see the test has two classes with the same times and are working, so I'm guessing its the latter case, but just wanted to double check.